app/src/main/java/com/kickstarter/viewmodels/CommentsViewHolderViewModel.kt

```diff
@@ -177,12 +187,72 @@ interface CommentsViewHolderViewModel {
             comment
                 .compose(takeWhen(this.onRetryViewClicked))
                 .compose(bindToLifecycle())
-                .subscribe(this.retrySendComment)
+                .subscribe {
+                    this.retrySendComment.onNext(it)
+                    this.commentCardStatus.onNext(CommentCardStatus.RE_TRYING_TO_POST)
+                }
+
+            val commentData = this.commentInput.map {
+                Pair(requireNotNull(it.comment?.body()), requireNotNull(it.project))
+            }
+
+            Observable
+                .combineLatest(commentData, postNewComment) { commentData, _ ->
+                    return@combineLatest this.apolloClient.createComment(
```

Comment:
you can extract the part inside the combineLatest into a function, something like: 

```
private fun executePostCommentMutation(commentData: Pair<String, Project>) =
            this.apolloClient.createComment(
                PostCommentData(
                    project = commentData.second,
                    body = commentData.first,
                    clientMutationId = null,
                    parentId = null
                )
            ).doOnError {
                this.commentCardStatus.onNext(CommentCardStatus.FAILED_TO_SEND_COMMENT)
            }
                .onErrorResumeNext(Observable.empty())
```

so it will end up in something like : 
<img width="795" alt="Screen Shot 2021-06-04 at 2 58 56 PM" src="https://user-images.githubusercontent.com/4083656/120866778-74ec7180-c545-11eb-8176-5883f10f8d00.png">